### ISSUE TITLE
Security Concern: GitHub Credentials Pass-through Attack Vector

### ISSUE THREAD

**Issue by Mike-EEE** [gitextensions/gitextensions#7706]
## Current behaviour

In attempting to invoke `GitHub` -> `View pull requests...` I am presented with a dialog box asking for my username, password, and 2FA code.  This is not the ideal security practice.  Applications should not ever ask for a password like this as this looks and feels like a password stealing attack, and can indeed be used for such purposes.

## Expected behaviour

Ideally, the application should open a web browser to a certificate-based resource so that the user can inspect its location and subsequent authenticity.  The results of this remote resource should set a local resource (e.g. cookie or token) which the application should then utilize for its own security context.   

There is also the use of [Personal Access Tokens](https://help.github.com/en/github/authenticating-to-github/creating-a-personal-access-token-for-the-command-line).  That way, the user can specifically create a token for Git Extensions and essentially run it under the context of the user, disabling it at a future time if a compromise has been detected.  This is much less disruptive than having to reset (and remember) yet another password.

## Steps to reproduce

Go to application menu found under application title bar, select `GitHub`, then select `View pull requests...`

## Screenshots

![](https://i.imgur.com/0VbyPcs.png)

## Did this work in previous version of GitExtensions

I do not recall ever being presented with username/password like this, and in fact recall using personal access tokens instead.  Paired with #7705 I have to wonder if something has indeed been compromised here. 😞

## Environment

- Git Extensions 3.3.1.7897
- Build 5a97671645532bcedc443bac7b727f40db47cb5c
- Git 2.25.0.windows.1
- Microsoft Windows NT 10.0.18363.0
- .NET Framework 4.8.4075.0
- DPI 96dpi (no scaling)

## Diagnostics

NA

**Comment by Mike-EEE** [comment]
The interesting aspect to note of course is that I am already connecting to GitHub through Git Extensions by way of Personal Access Token, as it is how I am push/pulling/etc.  It would seem that this would be leveraged somehow and only asking for further authentication if the current PAT doesn't have required permissions.

**Comment by gerhardol** [comment]
> Ideally, the application should open a web browser

This is how was before (kindof), which had other problems
Get your point though

**Comment by Mike-EEE** [comment]
Yeah @gerhardol I seem to recall something like that... it was IE-based or something and didn't work at all due to all the new fancy JavaScript. 😆

I am thinking if not web-based (because you would also have to consider and account for different browsers), then the Personal Access Tokens would be the better route, especially since they already seem to be used for git-related functionality.  

That is, I already have one created for (and by?) Git Extensions in my PAT list on GitHub.  Perhaps the answer here is to smooth out the experience there and leverage that as much as possible, especially if it is already being created by Git Extensions.
